src/PhpDoc/ObjectProphecy/TypeNodeResolverExtension.php

```diff
@@ -24,10 +24,7 @@
  */
 final class TypeNodeResolverExtension implements PhpDoc\TypeNodeResolverAwareExtension, PhpDoc\TypeNodeResolverExtension
```

Comment:
I would vote for removing this extension turning `ObjectProphecy|Foo` into `ObjectProphecy<Foo>`:

- defining the type as `ObjectProphecy|Foo` is a lie as the prophecy is **not** an instance of Foo
- phpstan would complain about the missing template type for `ObjectProphecy` anyway

Reply:
:+1: for going toward a generic, I already use them. We could release this as a new major to underline this.

Reply:
@Jean85 yeah I think make sense as a new Major so also consistent to the phpstan version.

Reply:
I removed at [4696fad](https://github.com/Jan0707/phpstan-prophecy/pull/348/commits/4696fad7ceaf31e567dfad8d83beea5bcfc3b66d)